starlite/contrib/sqlalchemy/base.py

```diff
@@ -0,0 +1,112 @@
+"""Application ORM configuration."""
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, TypeVar
+from uuid import UUID, uuid4
+
+from pydantic import AnyHttpUrl, AnyUrl, EmailStr
+from sqlalchemy import JSON, MetaData, String, Uuid
+from sqlalchemy.event import listens_for
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    Session,
+    declarative_mixin,
+    declared_attr,
+    mapped_column,
+    registry,
+)
+
+BaseT = TypeVar("BaseT", bound="Base")
+
+convention = {
+    "ix": "ix_%(column_0_label)s",
+    "uq": "uq_%(table_name)s_%(column_0_name)s",
+    "ck": "ck_%(table_name)s_%(constraint_name)s",
+    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    "pk": "pk_%(table_name)s",
+}
+"""Templates for automated constraint name generation."""
+
+
+@listens_for(Session, "before_flush")
+def touch_updated_timestamp(session: Session, *_: Any) -> None:
+    """Set timestamp on update.
+
+    Called from SQLAlchemy's
+    [`before_flush`][sqlalchemy.orm.SessionEvents.before_flush] event to bump the `updated`
+    timestamp on modified instances.
+
+    Args:
+        session: The sync [`Session`][sqlalchemy.orm.Session] instance that underlies the async
+            session.
+    """
+    for instance in session.dirty:
+        if hasattr(instance, "updated"):
+            instance.updated = datetime.now()
+
+
+@declarative_mixin
+class UUIDPrimaryKey:
+    """UUID Primary Key Field Mixin."""
+
+    __abstract__ = True
+
+    id: Mapped[UUID] = mapped_column(default=uuid4, primary_key=True)
+    """UUID Primary key column."""
+
+
+@declarative_mixin
+class AuditColumns:
+    """Created/Updated At Fields Mixin."""
+
+    __abstract__ = True
+
+    created: Mapped[datetime] = mapped_column(default=datetime.now)
+    """Date/time of instance creation."""
+    updated: Mapped[datetime] = mapped_column(default=datetime.now)
+    """Date/time of instance last update."""
+
+
+@declarative_mixin
+class CommonTableAttributes:
+    """Common attributes for SQLALchemy tables."""
+
+    __abstract__ = True
+    __name__: str
+    __table__: Any
+
+    # noinspection PyMethodParameters
+    @declared_attr.directive
+    def __tablename__(cls) -> str:  # pylint: disable=no-self-argument
+        """Infer table name from class name."""
+        return cls.__name__.lower()
```

Comment:
should we do CamelCase to snake_case for these do you think?

Reply:
Would certainly be useful, as that's the common naming scheme promoted by SQLA as well.

Reply:
Yes, I agree.  Snake case is the way to go for table names. I'll update this accoridngly.